plugins/widget/plugin.js

```diff
@@ -3078,28 +3071,46 @@
 		var changed = 0,
 			classes = getStyleClasses( style ),
 			updatedClasses = widget.data.classes || {},
-			cl;
+			cl,
+			styles = style.getDefinition().styles,
+			attributes = style.getDefinition().attributes;
 
 		// Ee... Something is wrong with this style.
-		if ( !classes )
+		if ( !classes && !styles && !attributes ) {
 			return;
+		}
 
-		// Clone, because we need to break reference.
-		updatedClasses = CKEDITOR.tools.clone( updatedClasses );
+		if ( classes ) {
+			// Clone, because we need to break reference.
+			updatedClasses = CKEDITOR.tools.clone( updatedClasses );
 
-		while ( ( cl = classes.pop() ) ) {
-			if ( apply ) {
-				if ( !updatedClasses[ cl ] )
-					changed = updatedClasses[ cl ] = 1;
-			} else {
-				if ( updatedClasses[ cl ] ) {
+			while ( ( cl = classes.pop() ) ) {
+				if ( apply ) {
+					if ( !updatedClasses[ cl ] )
+						changed = updatedClasses[ cl ] = 1;
+				} else if ( updatedClasses[ cl ] ) {
 					delete updatedClasses[ cl ];
 					changed = 1;
 				}
 			}
+			if ( changed ) {
+				widget.setData( 'classes', updatedClasses );
+			}
+		}
+
+		toggleStyleAttribute( widget.element, styles, 1, apply );
+		toggleStyleAttribute( widget.element, attributes, 0, apply );
+	}
+
+	function toggleStyleAttribute ( element, property, styleOrAttr, apply ) {
```

Comment:
Having a `Boolean` property with name like `styleOrAttr` is misleading. Without looking at function code is not possible to assume what `styleOrAttr=false` or `styleOrAttr=true` really means. But, it is also quite difficult to come up with proper name in such cases 🤔 

You may go same way as `apply` param - if `true` it means apply/set if not - remove. So here it would be `attributes` - `true` means apply attributes, false - styles (far from perfect IMHO).

The more readable way (but a little bit "inflated" too) would be something like:

```
function toogleAttributes( element, properties, apply ) { 
	toogleProperties( element, properties, apply, 'Attribute' );
}

function toogleStyles( element, properties, apply ) {
	toogleProperties( element, properties, apply, 'Style' );
}

function toogleProperties( element, properties, apply, propertyType ) { ... }
``` 
or actually, you may just stay with `toogleProperties` (skipping `toogleAttributes` and `toggleStyles`) and call it with proper params in the first place.

Reply:
One more thing, `property` param, should be `properties` as it is an object containing multiple properties.